### PR TITLE
[FLINK-16092][docs-zh] Translate dev/table/functions/index.zh.md into Chinese

### PR DESCRIPTION
**Pull request by zhule0601**
## What is the purpose of the change

This pull request translates the overview part of the function document into Chinese.

## Brief change log

  - Translate docs/dev/table/functions/index.zh.md into Chinese

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`:  no
  - The serializers:  no 
  - The runtime per-record code paths (performance sensitive):  no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper:  no 
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature?  no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: docs/dev/table/functions/index.zh.md [hunk 1]
```diff
@@ -24,74 +24,74 @@ specific language governing permissions and limitations
...
 
-Note that system functions always precede catalog's, and temporary functions always precede persistent on their own dimension
-in function resolution order explained below.
+1. 临时性系统函数
+2. 系统函数  
+3. 临时性 Catalog 函数
+4. Catalog 函数
 
-Referencing Functions
+请注意，系统函数始终优先于目录函数解析，临时函数始终优先于持久化函数解析，
```
Comment (libenchao) [inline docs/dev/table/functions/index.zh.md]: `目录函数`这里应该是手误吧，我看其他地方都没有翻译`Catalog`

File: docs/dev/table/functions/index.zh.md [hunk 2]
```diff
@@ -24,74 +24,74 @@ specific language governing permissions and limitations
...
+函数解析优先级如下所述。
+
+函数引用
 ---------------------
 
-There are two ways users can reference a function in Flink - referencing function precisely or ambiguously.
+用户在 Flink 中可以通过精确、模糊两种引用方式引用函数。
 
-## Precise Function Reference
+### 精确函数引用
```
Comment (libenchao) [inline docs/dev/table/functions/index.zh.md]: 看起来所有的标题都由 `##` 变成了 `###` ?

Comment (zhule0601) [reply docs/dev/table/functions/index.zh.md]: 这里的## 与父标题的形式太过接近，不修改为### 的话很难分辨，如图：
image

File: docs/dev/table/functions/index.zh.md (hunk 1 above)
Comment (zhule0601) [reply docs/dev/table/functions/index.zh.md]: 不好意思，这里手误了，我会修改回来的。

File: docs/dev/table/functions/index.zh.md (hunk 2 above)
Comment (libenchao) [reply docs/dev/table/functions/index.zh.md]: OK.

**Comment by libenchao** [comment]
LGTM, +1 for merging.
BTW, you don't need to `squash` your commits and force push, this will make reviewers difficult to track changes.

**Comment by zhule0601** [comment]
@libenchao Thanks for your review!  I'm still a beginner about translation, I will pay more attention next time.

**Comment by wuchong** [comment]
Thanks @zhule0601 and @libenchao , I will have a final look.

**Pull request closed without merging**